integration-tests/fixtures/generate-bundle-stats/astro/astro-integration.test.ts

```diff
@@ -0,0 +1,191 @@
+          );
+          const serverData = (await serverRes.json()) as { stats: string };
+          const serverStats = JSON.parse(serverData.stats) as unknown;
+
+          expect(serverStats).toMatchSnapshot({
+            builtAt: expect.any(Number),
+            duration: expect.any(Number),
+            outputPath: expect.stringContaining("dist"),
+            bundleName: expect.stringContaining(serverBundleName),
+            plugin: {
+              name: expect.stringMatching("@codecov/astro-integration"),
+            },
+            assets: expect.arrayContaining([
+              {
+                name: expect.any(String),
+                normalized: expect.any(String),
+                size: expect.any(Number),
+                gzipSize: expect.anything(),
+              },
+            ]),
+            chunks: expect.arrayContaining([
+              {
+                id: expect.any(String),
+                uniqueId: expect.any(String),
+                initial: expect.any(Boolean),
+                entry: expect.any(Boolean),
+                files: expect.arrayContaining([expect.any(String)]),
+                names: expect.arrayContaining([expect.any(String)]),
+              },
+            ]),
+            modules: expect.arrayContaining([
+              {
+                name: expect.any(String),
+                size: expect.any(Number),
+                chunkUniqueIds: expect.arrayContaining([expect.any(String)]),
+              },
+            ]),
+          });
+        },
+        { timeout: 25_000 },
+      );
+    });
+
+    describe("invalid bundle name is passed", () => {
+      beforeEach(async () => {
+        const config = new GenerateConfig({
+          plugin: "astro",
+          configFileName: "astro",
+          format: "esm",
+          detectFormat: "esm",
+          version: `v4`,
+          detectVersion: "v4",
+          file_format: "mjs",
+          enableSourceMaps: false,
+          overrideOutputPath: `${astroApp}/astro.config.mjs`,
+        });
+
+        await config.createConfig();
+        config.removeBundleName(`test-astro-v${version}`);
+        await config.writeConfig();
+      });
+
+      afterEach(async () => {
+        await $`rm -rf ${astroApp}/astro.config.mjs`;
+        await $`rm -rf ${astroApp}/dist`;
+        await $`rm -rf ./fixtures/generate-bundle-stats/astro/dist`;
+      });
+
+      it(
+        "warns users and exits process with a code 1",
+        async () => {
+          const id = `astro-v${version}-${Date.now()}`;
+          const API_URL = `http://localhost:8000/test-url/${id}/200/false`;
+
+          // prepare and build the app
+          const { exitCode, stdout } =
+            await $`cd test-apps/astro && API_URL=${API_URL} pnpm run build`.nothrow();
+
+          expect(exitCode).toBe(1);
+          // for some reason this isn't being outputted in the test env
+          expect(stdout.toString()).toContain(
```

Comment:
>          // for some reason this isn't being outputted in the test env

Could it be that we are outputting to `stderr` instead of `stdout`? Just a guess.

Reply:
I think that's a left over comment that's just been copy pasted over time, before i believe we were skipping that test, but as you can see here, we're not